### PR TITLE
Add option for specifying config directly as JSON

### PR DESCRIPTION
**Pull request by bgood**
This PR adds new variables for basic settings to be passed into the Cloud Scheduler module. The new variables provide an easy way for users to specify common settings when managing a single spanner instance.

The PR also adds a variable to directly specify JSON that should be used by Cloud Scheduler.  This variable allows users to define configuration for multiple spanners and set configuration values that are directly enabled via Terraform variables.

### CODE REVIEW

**Comment by bgood** [comment]
This PR resolves Issue #83

File: terraform/modules/scheduler/variables.tf [hunk 1]
```diff
@@ -62,3 +86,9 @@ variable "state_spanner_name" {
   nullable = true
   default  = null
 }
+
+variable "json_config" {
+  type        = string
+  default     = ""
+  description = "Base 64 encoded json that is the autoscaler configuration for the Cloud Scheduler payload. Using this allows for setting autoscaler configuration for multiple spanners and parameters that are not directly exposed through variables."
```
[the remark this answers is not in the document]
Comment (henrybell) [reply terraform/modules/scheduler/variables.tf:93]: I meant more, can we take care of the `base64encode(jsonencode([{...` in the module, and not require that the user (calling the module) has to do this in "their" code?

[the remark this answers is not in the document]
Comment (bgood) [reply terraform/modules/scheduler/variables.tf:93]: Gotcha.

To do that, I think we would need to change the input variable to a list of objects.  When using objects as input variables, we would then need to define the keys and types of the object (the config). That feels a bit heavy to me vs the `base64encode(jsonencode([{...`. 

Thoughts?

File: terraform/modules/scheduler/main.tf [hunk 2]
```diff
@@ -14,28 +14,16 @@
...
       merge ({
         "projectId" : "${var.project_id}",
         "instanceId" : "${var.spanner_name}",
         "scalerPubSubTopic" : "${var.target_pubsub_topic}",
-        "units" : "PROCESSING_UNITS",
-        "minSize" : 100,
-        "maxSize" : 2000,
-        "scalingMethod" : "LINEAR",
+        "units" : "${var.units}",
+        "minSize" : ${var.min_size},
```
[the remark this answers is not in the document]
Comment (bgood) [reply terraform/modules/scheduler/main.tf:24]: Do you mean the underscores or the lack of quotes?

[the remark this answers is not in the document]
Comment (henrybell) [reply terraform/modules/scheduler/main.tf:24]: Lack of quotes

[the remark this answers is not in the document]
Comment (henrybell) [reply terraform/modules/scheduler/main.tf:24]: I think the interpolation would need to be inside a (quoted) string, or use `var.min_size`? The above code errors for me, perhaps I am missing something.

File: terraform/modules/scheduler/variables.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (henrybell) [reply terraform/modules/scheduler/variables.tf:93]: I would need to write it out to get the feel of it... if the alternative is no better, let's stick with the above and the added example.

**Review by henrybell — APPROVED** [review]

**Pull request merged**